File: src/p0409/python/solution.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        m = dict()
        for c in s:
            if c in m:
                m[c] += 1
            else:
                m[c] = 1
        ans = 0
        for v in m.values():
            ans += v // 2 * 2
            if v % 2 == 1 and ans % 2 == 0:
                ans += 1
                
        return ans

File: src/p0409/golang/solution.go
func longestPalindrome(s string) int {
	m := make(map[byte]int)
	for _, c := range s {
		m[byte(c)] += 1
	}
	var (
		largestKey byte
		largestOdd int
		ans        int
	)
	for k, v := range m {
		if v%2 == 1 && v > largestOdd {
			largestOdd = v
			largestKey = k
		}
	}
	ans = largestOdd
	delete(m, largestKey)
	for _, v := range m {
		if v%2 == 0 {
			ans += v
		} else if largestOdd != 1 && v != 1 {
			ans += v - 1
		}
	}

	return ans
}

// 腾讯